backend/tests/unit/modules/extract/test_extract.py

```diff
@@ -0,0 +1,159 @@
+    DataValidationError,
+    FileExtensionNotSupportedError,
+)
+from modules.extract.domain.ports import repositories, units_of_work
+from modules.extract.services.commands import extract
+from tests import test_const
+
+from ....common.annotations import YieldFixture
+from .fakers import (
+    TestExtractDomainRepository,
+    TestExtractUnitOfWork,
+    TestFileDomainRepository,
+)
+
+
+def test_extract_successfully_read_csv_file(
+    test_data_unit_of_work: units_of_work.AbstractExtractUnitOfWork,
+    test_extract_repository: repositories.AbstractExtractDomainRepository,
+    test_file_repository: repositories.AbstractFileDomainRepository,
+):
+    # Given
+    test_dataset_size = 10
+
+    # When
+    input_data: value_objects.InputData = extract(
+        extract_unit_of_work=test_data_unit_of_work,
+        command=commands.ExtractData(
+            file=test_const.CORRECT_INPUT_CSV.read_bytes(),
+            file_name=test_const.CORRECT_INPUT_CSV.name,
+        ),
+    )
+
+    # Then
+    assert not input_data.empty  # type: ignore[attr-defined]
+    assert input_data.shape[0] == test_dataset_size  # type: ignore[attr-defined]
+    assert test_file_repository.file_exists(test_const.CORRECT_INPUT_CSV)
+    assert len(test_extract_repository.list()) == 1  # type: ignore[attr-defined]
+    assert (
+        test_extract_repository.list()[0].input_file_name  # type: ignore[attr-defined]
+        == test_const.CORRECT_INPUT_CSV.name
+    )
+    assert (
+        test_extract_repository.list()[0].saved_file_name  # type: ignore[attr-defined]
+        == test_const.CORRECT_INPUT_CSV.name
+    )
+
+
+def test_extract_successfully_read_xlsx_file(
+    test_data_unit_of_work: units_of_work.AbstractExtractUnitOfWork,
+    test_extract_repository: repositories.AbstractExtractDomainRepository,
+    test_file_repository: repositories.AbstractFileDomainRepository,
+):
+    # Given
+    test_dataset_size = 10
+
+    # When
+    input_data: value_objects.InputData = extract(
+        extract_unit_of_work=test_data_unit_of_work,
+        command=commands.ExtractData(
+            file=test_const.CORRECT_INPUT_XLSX.read_bytes(),
+            file_name=test_const.CORRECT_INPUT_XLSX.name,
+        ),
+    )
+    # Then
+    assert not input_data.empty  # type: ignore[attr-defined]
+    assert input_data.shape[0] == test_dataset_size  # type: ignore[attr-defined]
+    assert test_file_repository.file_exists(test_const.CORRECT_INPUT_XLSX)
+    assert len(test_extract_repository.list()) == 1  # type: ignore[attr-defined]
+    assert (
+        test_extract_repository.list()[0].input_file_name  # type: ignore[attr-defined]
+        == test_const.CORRECT_INPUT_XLSX.name
+    )
+    assert (
+        test_extract_repository.list()[0].saved_file_name  # type: ignore[attr-defined]
+        == test_const.CORRECT_INPUT_XLSX.name
+    )
+
+
+def test_extract_successfully_read_xls_file(
+    test_data_unit_of_work: units_of_work.AbstractExtractUnitOfWork,
```

Comment:
There MUST BE NOT fixtures for domain repositories, only unit of work can be used.
So please remove domain repositories fixtures.

Queries must be done with query repositories.
Please, replace thew with that.